packages/gradle-plugin/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/AgpConfiguratorUtils.kt

```diff
@@ -104,3 +107,11 @@ fun getPackageNameFromManifest(manifest: File): String? {
     return null
   }
 }
+
+fun getHostIpAddress(): String =
+  NetworkInterface.getNetworkInterfaces().asSequence()
+    .filter { it.isUp && !it.isLoopback }
+    .flatMap { it.inetAddresses.asSequence() }
+    .filter { it is Inet4Address && !it.isLoopbackAddress }
+    .map { it.hostAddress }
```

Comment:
This seems like the correct approach. 

The only thing to be aware is that this should not prevent implementing Gradle configuration caching in the future: https://github.com/facebook/react-native/issues/45154

Reply:
@cortinico  I'm not sure I understand... Does this need additional changes or is it ok like this?

Reply:
It requires extra caution (it's a reminder for myself) as this could prevent Gradle Configuration caching alltogether.